public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -24,22 +25,53 @@
 
     @Autowired
     private ConceptDao conceptDao;
-
     @Autowired
-    private AnalysisResultDao analysisResultDao;
-
+    private QuestionConceptDao  questionConceptDao;
     @Autowired
     private AchillesAnalysisDao achillesAnalysisDao;
-
+    @Autowired
+    private AchillesResultDao achillesResultDao;
     @Autowired
     private DbDomainDao dbDomainDao;
+    @Autowired
+    private EntityManager entityManager;
 
     public static final long PARTICIPANT_COUNT_ANALYSIS_ID = 1;
     public static final long COUNT_ANALYSIS_ID = 3000;
     public static final long GENDER_ANALYSIS_ID = 3101;
     public static final long AGE_ANALYSIS_ID = 3102;
+    public static final long SURVEY_COUNT_ANALYSIS_ID = 3100;
+    public static final long SURVEY_GENDER_ANALYSIS_ID = 3111;
+    public static final long SURVEY_AGE_ANALYSIS_ID = 3112;
+
+    public static Map<String, String> ageStratumNameMap  = new HashMap<String, String>();
+    public static Map<String, String> genderStratumNameMap = new HashMap<String, String>();
+
+
+    public void setAgeStratumNameMap() {
+        ageStratumNameMap.put("1", "0-18 yrs old");
+        ageStratumNameMap.put("2", "18-29 yrs old");
+        ageStratumNameMap.put("3", "30-39 yrs old");
+        ageStratumNameMap.put("4", "40-49 yrs old");
+        ageStratumNameMap.put("5", "50-59 yrs old");
+        ageStratumNameMap.put("6", "60-69 yrs old");
+        ageStratumNameMap.put("7", "70-79 yrs old");
+        ageStratumNameMap.put("8", "80-89 yrs old");
+        ageStratumNameMap.put("9", "90-99 yrs old");
+        ageStratumNameMap.put("10", "100-109 yrs old");
+        ageStratumNameMap.put("11", "110-119 yrs old");
+        ageStratumNameMap.put("12", "120-129 yrs old");
+        ageStratumNameMap.put("13", "130-139 yrs old");
+    }
 
-
+    public void setGenderStratumNameMap() {
+        /* This is to slow to use the db */
```

Comment:
So joining to concept for the relevant queries is too slow? Can you give an example of how much this speeds things up? Is this only a problem because you're doing O(n) lookups below?

(I ask in part because we have the same situation in cohort review... @freemabd1 may be interested.)

Reply:
So, intially i wanted to query this once and make a static map in the constructor .  But it would not work. I'm not sure how the lifecyle of the controller is between requests.  Ideally the map could be queried once  and live for subsequent requests to the api.  Barring that I added the query in the endpoint in the controller. It added a full second on each request to that endpoint.